src/api/nodeGrouping/types.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * Copyright Linkurious SAS 2012 - 2024
+ *
+ * - Created on 2024-02-06.
+ */
+
+export interface NodeGroupingRule {
+  id: number;
+  sourceKey: string;
+  name: string;
+  itemType: string;
```

Comment:
`itemType` and `propertyKey` are very specific to how the first mvp of node grouping works, but will not adapt well to future improvement to node-grouping. Can you suggest a more generic way to represent the grouping rule that will allow for future interations that will involve different kinds of grouping rules?